### DISCUSSION TITLE
How to customize account-ui components in KC 25?

### DISCUSSION THREAD

**Discussion by jperlak** [keycloak/keycloak#31966] [Q&A]
Keycloak 25 introduced new account-console theme (keycloak.v3). Unfortunately, now I do not see any possibility to customize the layouts like the `PersonalInfo` view.
In project I am working on, we want to change the layout a little bit, e.g. use table tags (like `th`, `td`) instead of only divs (requirement of fulfilling the WCAG standard).

Since everything is in the `account-ui` I don't think is as easy to customize as it was before, e.g. in KC 23 I would just add my `account/resources/content/account-page/AccountPage.js` file and customize whatever I needed. Now I managed to just apply my custom CSS & logotype.

Is there any possibility to customize such views now?
Or have anyone come up with some workaround?
Or maybe there is some sort of plans to give the possibility to customize/overwrite the account-ui components?

**Comment by darius-m** [comment]
You can still customize things as you did before (especially the logo), by deriving the account v3 theme. However, this is complicated by the fact that the account UI is built using Vite (I'm guessing it is responsible for the file names in the end?) and the files have random artifacts in the name (it's done to make sure the new version is used when rebuilding the pages), so you have to update the file names to match the ones in the release every time. Also, the JavaScript files are minified, so you pretty much need to set up the whole build process.

You can start from https://github.com/keycloak/keycloak-quickstarts/tree/latest/extension/extend-account-console and copying the `account-ui` files, but it honestly feels a bit too involved for simple patches.

**Reply by jperlak** [reply]
I have already changed the logo & favicon, as well as the CSS (the PF5 update was quite easy to handle). However, I struggle with customizing the, let's say, built-in components like [PersonalInfo](https://github.com/keycloak/keycloak/blob/release/25.0/js/apps/account-ui/src/personal-info/PersonalInfo.tsx).

Unfortunately, I have already tried the extend-account-console guide, but it's either not what I need or I do not understand it. As far as I understand it's about styling (which I've already changed) or adding new pages (which I'm not interested in). What I need is modyfing the account-ui components but the problem is, as you've noticed, with the building process which seems to be handled in a different way than in past releases like KC 23.

Anyway, what do you mean by `and copying the account-ui files`? Do you have some sort of working example (or could you provide even a small one) for editing such files? I'm asking since I think I've tested different approaches which were fine in previoes releases and none of them is working now (my files for account theme are not overwriting the KC ones as they once were).

**Reply by darius-m** [reply]
The files for the account UI are here https://github.com/keycloak/keycloak/tree/main/js/apps/account-ui (sorry, I assumed you knew where they were when you mentioned `account-ui`). The directory contains some instructions on how you can fetch the theme and build it using `pnpm` (I assume `npm` works as well).

I don't have a working example at the moment, sorry. However, from previous experience, you can probably start from a `node`-based container, or a different container (e.g., the RedHat UBI container) and install `node` and `npm`, install the theme and build it. You can change the theme and build it yourself to fetch the artifacts. Building can also be done in a CI/CD environment (e.g., using GitHub actions or GitLab pipelines).
